tests/acceptance/main-test.js

```diff
@@ -62,6 +62,18 @@ module('Acceptance | main', function(hooks) {
 
       assert.equal(user.get('profile.firstName'), 'Grace');
       assert.equal(user.get('profile.lastName'), 'Hopper');
+
+      let profile = this.store.createRecord('profile', { firstName: 'Terry', lastName: 'Bubblewinkles' });
+      changeset.set('profile', profile);
```

Comment:
This is the important test that shows setting a new belongsTo relationship.  Moreover, nested key setters still work.